### PR TITLE
Major update to backup/restore

### PR DESCRIPTION
**Pull request by aanthony1243**
-- default backup format is compatible with enterprise backup/restore
  -- live backup/restore functionality
  -- timestamp-based start/end filtering arguments
  -- previous functionality preserved under -legacy flag

###### Required for all non-trivial PRs
- [X] Rebased/mergable
- [ ] Tests pass
- [X] CHANGELOG.md updated

###### Required only if applicable
_You can erase any checkboxes below this note if they are not applicable to your Pull Request._
- [ ] [InfluxData Documentation](https://github.com/influxdata/docs.influxdata.com): issue filed or pull request submitted \<link to issue or pull request\>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,7 +5,9 @@
 * The default logging format has been changed. See [#9055](https://github.com/influxdata/influxdb/pull/9055) for details.
 
 ### Features
-
+- [#9146](https://github.com/influxdata/influxdb/issues/9146): Backup/Restore can produce/consume data in the same format as the enterprise backup/restore tool. 
+- [#8880](https://github.com/influxdata/influxdb/issues/8879): Restore runs in online mode
+- [#8879](https://github.com/influxdata/influxdb/issues/8879): Export functionality using start/end to filter exported data by timestamp
```
Comment (joelegasse) [inline CHANGELOG.md]: Was it necessary that all of these changes be included in a single pull request? If not, I would prefer to see each feature have a separate PR. Combining them makes it more difficult to review and know which changes are in support of which feature.

File: tsdb/engine/tsm1/engine.go [hunk 2]
```diff
@@ -690,6 +690,129 @@ func (e *Engine) Backup(w io.Writer, basePath string, since time.Time) error {
...
+		}
+	}
+
+	err = w.WriteIndex()
+	if err != nil {
+		return err
+	}
+
+	// make sure the whole file is out to disk
+	w.Flush()
```
Comment (jwilder) [inline tsdb/engine/tsm1/engine.go]: Missing `err` check.

File: tsdb/engine/tsm1/engine.go [hunk 3]
```diff
@@ -690,6 +690,129 @@ func (e *Engine) Backup(w io.Writer, basePath string, since time.Time) error {
...
+		if minTime >= start && minTime <= end ||
+			maxTime >= start && maxTime <= end ||
+			minTime <= start && maxTime >= end {
+			err := w.WriteBlock(key, minTime, maxTime, buf)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
```
Comment (jwilder) [inline tsdb/engine/tsm1/engine.go]: This needs a check for iteration errors:

```
	if err := bi.Err(); err != nil {
		return err
	}
```

File: tsdb/engine/tsm1/engine.go [hunk 4]
```diff
@@ -747,8 +870,10 @@ func (e *Engine) overlay(r io.Reader, basePath string, asNew bool) error {
 		tr := tar.NewReader(r)
 		for {
 			if fileName, err := e.readFileFromBackup(tr, basePath, asNew); err == io.EOF {
+				fmt.Println(err)
```
Comment (jwilder) [inline tsdb/engine/tsm1/engine.go]: Remove

File: tsdb/engine/tsm1/engine.go [hunk 5]
```diff
@@ -747,8 +870,10 @@ func (e *Engine) overlay(r io.Reader, basePath string, asNew bool) error {
...
 				break
 			} else if err != nil {
+				fmt.Println(err)
```
Comment (jwilder) [inline tsdb/engine/tsm1/engine.go]: Remove

**Pull request closed without merging**